filebeat/module/postgresql/log/ingest/pipeline.yml

```diff
@@ -41,15 +47,22 @@ processors:
     field: event.type
     value:
       - info
+    if: "ctx?.postgresql?.log?.sql_state_code == null || (ctx.postgresql.log.sql_state_code ==~ /^0[012].*/)"
 - append:
```

Comment:
The [`set` processor](https://www.elastic.co/guide/en/elasticsearch/reference/7.11/set-processor.html) could be used to add `event.type` instead of `append`, so it is not set as an array in the document.